File: nutritionLabJava/FoodItem.java
/**
 * Represents a food item with nutritional information (fat, carbs, protein).
 * Provides methods to calculate calories and print info.
 */
public class FoodItem {
    // Name of the food
    private String name;
    // Amounts per serving
    private double fat;
    private double carbs;
    private double protein;

    /**
     * Default constructor.
     * Initializes the item as Water with zero fat, carbs, and protein.
     */
    public FoodItem() {
        this.name    = "Water";
        this.fat     = 0.0;
        this.carbs   = 0.0;
        this.protein = 0.0;
    }

    /**
     * Parameterized constructor.
     * @param name      Name of the food item
     * @param fat       Grams of fat per serving
     * @param carbs     Grams of carbohydrates per serving
     * @param protein   Grams of protein per serving
     */
    public FoodItem(String name, double fat, double carbs, double protein) {
        this.name    = name;
        this.fat     = fat;
        this.carbs   = carbs;
        this.protein = protein;
    }

    /**
     * Calculates calories for a given number of servings.
     * Formula: (fat * 9 + carbs * 4 + protein * 4) * numServings
     * @param numServings Number of servings
     * @return Total calories
     */
    public double getCalories(double numServings) {
        return ((fat * 9) + (carbs * 4) + (protein * 4)) * numServings;
    }

    /**
     * Prints the nutritional information per serving to the console.
     */
    public void printInfo() {
        System.out.printf("Nutritional information per serving of %s:%n", name);
        System.out.printf("  Fat: %.2f g%n", fat);
        System.out.printf("  Carbohydrates: %.2f g%n", carbs);
        System.out.printf("  Protein: %.2f g%n", protein);
    }
}

File: nutritionLab/FoodItem.cpp
#include "FoodItem.h"
#include <iostream>
#include <iomanip> 



// Define default constructor\
//if foodItem object is defined with no parameters these are the default values. Instead of name = "noName" use initializaition form just need a colon :
FoodItem::FoodItem(): name("Water"), fat(0), carbs(0), protein(0){}

FoodItem::FoodItem(std::string foodName, double foodFat, double foodCarbs, double foodProteins): name(foodName), fat(foodFat), carbs(foodCarbs), protein(foodProteins){}


std::string FoodItem::GetName() {
   return name;
}

double FoodItem::GetFat() {
   return fat;
}

double FoodItem::GetCarbs() {
   return carbs;
}

double FoodItem::GetProtein() {
   return protein;
}

double FoodItem::GetCalories(double numServings) {
   // Calorie formula
   double calories = ((fat * 9) + (carbs * 4) + (protein * 4)) * numServings;
   return calories;
}

void FoodItem::PrintInfo() {
    std::cout << std::fixed << std::setprecision(2);
    std::cout << "Nutritional information per serving of " << name  << ":" << std::endl;
    std::cout << "  Fat: " << fat << " g" << std::endl;
    std::cout << "  Carbohydrates: " << carbs << " g" << std::endl;
    std::cout << "  Protein: " << protein << " g" << std::endl;
}